File: ExcelDivider/src/test/java/ru/tsystems/divider/context/InMemoryTask.java
package ru.tsystems.divider.context;

import ru.tsystems.divider.dao.api.TaskDao;
import ru.tsystems.divider.entity.Task;

public class InMemoryTask extends InMemoryDaoGeneral<Task> implements TaskDao, ContextSimmulator<Task> {
    private static InMemoryTask dao;

    protected static InMemoryTask getInMemoryDao() {
        if (dao == null) {
            dao = new InMemoryTask();
        }
        return dao;
    }

    protected InMemoryTask() {
        super();
        reset();
    }

    /**
     * Finding task by key
     *
     * @param key Key
     * @return {@link Task}
     */
    @Override
    public Task getBykey(String key) {
        for (Task task : database) {
            if (key.equals(task.getKeys())) {
                return task;
            }
        }
        return null;
    }

    @Override
    public void reset() {

    }

    @Override
    public void setValue(String key, Task value) {

    }

    @Override
    public Task getValue(String key) {
        return null;
    }
}

File: ExcelDivider/src/test/kt/ru/tsystems/divider/context/InMemoryTask.kt
package ru.tsystems.divider.context

import ru.tsystems.divider.dao.api.TaskDao
import ru.tsystems.divider.entity.Task

object InMemoryTask : InMemoryDaoGeneral<Task>(), TaskDao, ContextSimmulator<Task> {

    init {
        reset()
    }

    /**
     * Finding task by key
     *
     * @param key Key
     * @return [Task]
     */
    override fun getBykey(key: String): Task? {
        for (task in database) {
            if (key == task.keys) {
                return task
            }
        }
        return null
    }
}